infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/SwitchLocationTrackLink.kt

```diff
@@ -4,7 +4,48 @@ import fi.fta.geoviite.infra.common.DomainId
 import fi.fta.geoviite.infra.common.IntId
 import fi.fta.geoviite.infra.common.JointNumber
 
+fun getLocationTrackDuplicateOfStatus(
+    mainTrack: LocationTrack,
+    mainAlignment: LayoutAlignment,
+    duplicatedTrack: LocationTrack,
+    duplicatedAlignment: LayoutAlignment,
+): LocationTrackDuplicate {
+    val mainTrackJoints = collectSwitchJoints(mainTrack, mainAlignment)
+    val duplicatedTrackJoints = collectSwitchJoints(duplicatedTrack, duplicatedAlignment)
+    val statuses = getDuplicateMatches(
+        duplicatedTrackJoints,
+        mainTrackJoints,
+        duplicatedTrack.id,
+        duplicatedTrack.duplicateOf,
+    )
+    return statuses.map { (jointIndex, status) ->
+        jointIndex to LocationTrackDuplicate(
+            duplicatedTrack.id as IntId,
+            duplicatedTrack.trackNumberId,
+            duplicatedTrack.name,
+            duplicatedTrack.externalId,
+            status,
+        )
+    }.first().second // There has to at least one found, since we know the duplicateOf is set
```

Comment:
Olisiko jopa `.single()` oikein? Kuulostaisi väärältä että näitä linkkejä löytyisi useampi kuin yksi silloin kun rajataan jo valmiiksi tiedossa olevilla raiteilla.

Reply:
Luulisin että se on edelleen teoriassa mahdollista jos yhdellä raiteella on useampia duplikaattiosio. Käytännössä noita ei pitäisi olla mutta jos sellainen tulee, en halua siitä 500-erroria.